### PR TITLE
[swift-ide-test] Add -enable-upcoming-feature ImportObjcForwardDeclarations support

### PR DESCRIPTION
**Pull request by NuriAmari**
In this first commit we add support for the -enable-upcoming-feature flag in general.

In the second commit we make sure that the ClangImporter constructed for use in swift-ide-test is configured appropriately when the ImportObjcForwardDeclarations upcoming language feature is enabled, or the language version is recent enough to enable it by default.

### CODE REVIEW

File: tools/swift-ide-test/swift-ide-test.cpp [hunk 1]
```diff
@@ -413,6 +413,9 @@ static llvm::cl::opt<bool> CodeCompleteCallPatternHeuristics(
         "Use heuristics to guess whether we want call pattern completions"),
     llvm::cl::cat(Category));
 
+// This option and the upcoming language feature ImportObjcForwardDeclarations
+// enable the same behavior. If we no longer rely upon
+// -enable-objc-forward-declarations, we may remove the option.
```
Comment (NuriAmari) [inline tools/swift-ide-test/swift-ide-test.cpp:418]: IIUC swift-ide-test is supposed to be an "internal tool", so its options aren't guaranteed to be stable. I have no idea who uses this option in other projects, but if they are migrated, the flag can be deleted.

**Comment by NuriAmari** [comment]
@swift-ci Please test

File: tools/swift-ide-test/swift-ide-test.cpp (hunk 1 above)
Comment (drodriguez) [reply tools/swift-ide-test/swift-ide-test.cpp:418]: I did not see this one in my compiled 5.9 `swift-ide-test`. Did I miss it or does it appear after that branch?

In any case, I 100% for removing this, if there's an alternate spelling which is closer to the compiler actualy spelling.

**Review by drodriguez — APPROVED** [review]
Seeing that `ImportObjcForwardDeclarations` needs special code to handle the new flag, I wonder if other upcoming features will also need special code to handle things. It is not the best situation, but each upcoming feature will need to do their due diligence to be supported by `swift-ide-test` (some of them might not even be "exposed" by `swift-ide-test` anyway).

File: tools/swift-ide-test/swift-ide-test.cpp (hunk 1 above)
Comment (NuriAmari) [reply tools/swift-ide-test/swift-ide-test.cpp:418]: It's there for me 🤷

Comment (ahoppen) [reply tools/swift-ide-test/swift-ide-test.cpp:418]: If tests pass with the flag removed, we can remove it. `swift-ide-test` is mainly intended for testing purposes.

**Review by ahoppen — APPROVED** [review]

**Comment by NuriAmari** [comment]
Updated to remove the old flag

**Comment by NuriAmari** [comment]
@swift-ci Please test

**Pull request merged**